pkgs/development/python-modules/torch/mk-torch-compile-check.nix

```diff
@@ -20,19 +20,41 @@ in
       "${lib.getBin stdenv.cc}/bin"
     ];
   }
-  ''
-    import torch
+  (
+    ''
+      import torch
 
 
-    @torch.compile
-    def opt_foo2(x, y):
-        a = torch.sin(x)
-        b = torch.cos(y)
-        return a + b
+    ''
+    +
+      # torch.compile requires OpenMP which is not available on Darwin
+      lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
+        @torch.compile
+      ''
+    + ''
+      def opt_foo2(x, y):
+          a = torch.sin(x)
+          b = torch.cos(y)
+          return a + b
 
 
-    print(
-      opt_foo2(
-        torch.randn(10, 10${deviceStr}),
-        torch.randn(10, 10${deviceStr})))
-  ''
+      print(
+        opt_foo2(
+          torch.randn(10, 10${deviceStr}),
+          torch.randn(10, 10${deviceStr})))
+    ''
+    +
+      # MPS is built by default on Darwin, so test it.
+      lib.optionalString stdenv.hostPlatform.isDarwin ''
+        assert torch.backends.mps.is_built(), "PyTorch not build with MPS enabled"
+        if not torch.backends.mps.is_available():
+            print("MPS not available because the current MacOS version is not 12.3+ "
+                  "and/or you do not have an MPS-enabled device on this machine.")
+
+        else:
+            print(
+              opt_foo2(
+                torch.randn(10, 10, device="mps"),
+                torch.randn(10, 10, device="mps")))
```

Comment:
I think it'd make more sense to model `mps` as one of the values for the `feature` argument. Then we add one more passthru test. IMO it's OK to add it unconditionally as long as we set its meta platforms to darwin

Reply:
I was actually also wondering that is the point of this mk-torch-compile-check.nix to actually do check if the @torch.compile function decorator works? It doesn't work on Darwin because there is no OpenMP supported by default llvm compiler.

Maybe the test for darwin should be in completely separate file.